Geopackage vocabulary harmonization #60990

Snegovoy98 opened the pull request:
There were renamed fields which was shown on the screenshots.

benoitdm-oslandia committed:
chore: fix prepare_commit to catch all modified/added files

benoitdm-oslandia committed:
fix(qgstest): in checkLongStr add ^ to locate where is the first diff

lbartoletti committed:
feat(python): Add an option to disable Python

lbartoletti committed:
chore(QgisApp): Add an enum to handle QgisApp boolean parameters

lbartoletti committed:
chore(main): typo, replace python by Python

Co-authored-by: Nyall Dawson <nyall.dawson@gmail.com>

gacarrillor committed:
[ux] Show line drop indicator instead of box for stacked diagram items (items don't have children so far),  because of that, we can drop at any index as well

gacarrillor committed:
[ux] Stacked diagrams: avoid items being appended to the tree view when dropped 'inside' an item; only enable drops on invalid items, which eases dropping handling

ptitjano committed:
qgselevationprofilecanvas: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the vertices of the trace.

This is mainly achieved by adding
`QgsElevationProfileCanvas::setSubsectionsSymbol()`. This method
allows to set the line symbol of the vertical lines. Then, It sets the
subsections symbol of the profile renderer of the plot item. It the
renderer does not exist yet, the subsections symbol is set when a new
profile renderer is created.

ptitjano committed:
qgselevationprofilecanvas: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the vertices of the trace.

This is mainly achieved by adding
`QgsElevationProfileCanvas::setSubsectionsSymbol()`. This method
allows to set the line symbol of the vertical lines. Then, It sets the
subsections symbol of the profile renderer of the plot item. It the
renderer does not exist yet, the subsections symbol is set when a new
profile renderer is created.

ptitjano committed:
qgselevationprofilecanvas: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the vertices of the trace.

This is mainly achieved by adding
`QgsElevationProfileCanvas::setSubsectionsSymbol()`. This method
allows to set the line symbol of the vertical lines. Then, It sets the
subsections symbol of the profile renderer of the plot item. It the
renderer does not exist yet, the subsections symbol is set when a new
profile renderer is created.

ptitjano committed:
qgselevationprofilewidget: Add actions for subsections indicator

This introduces two new actions in the options menu:
- an action to display or the hide the subsections indicator
- an action to to set the lines' symbology.

ptitjano committed:
qgselevationprofilewidget: Add actions for subsections indicator

This introduces two new actions in the options menu:
- an action to display or the hide the subsections indicator
- an action to to set the lines' symbology.

nyalldawson committed:
Update src/app/qgisapp.h

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

jef-n committed:
Allow free naming of project properties

Allows:

<properties name="properties">
  <properties name="1">
    <properties name="2" type="QString">foo</properties>
  </properties>
</properties>

Data-driven properties names could violate that monstrous regular
expression earlier (eg. using layer ids with leading digits as key was
illegal) and trigger "Entry token invalid" for no too obvious reason
(even when trying to remove a non-existing entry).

Turns

<properties>
  <foo>
    <bar type="QString">baz</bar>
  </foo>
</properties>

into

<properties name="properties">
  <properties name="foo">
    <properties name="bar" type="QString">baz</properties>
  </properties>
</properties>

agiudiceandrea committed:
Fix Value Map widget with NULL value and field length < 38

troopa81 committed:
refactor(ServerProperties): Create a server properties widget

And reuse it in vector, raster and vector tile properties widget

troopa81 committed:
refactor(ServerProperties): write XML in the server properties class

instead of writing it in QgsMapLayer

gacarrillor committed:
[ux] Avoid copy on drag and drop by correctly setting the current index to span rows

gacarrillor committed:
[ux] Stacked diagrams: correct order in multiple-item drag and drop

Withalion committed:
Fix line editing tool crash when used wrong

Withalion committed:
Disable editing tool, when disabling editing in 3D

Withalion committed:
Add minimum size for paintbrush 3D editing tool

Withalion committed:
Fix paintbrush editing tool movement

Withalion committed:
Fix build issue & crash on closing 3D window

Fixes msvc build issue caused by uniqueqobject pointer, we revert back
to raw pointers. Fixes QGIS crash on 3D window closed when editing tools
are selected. We can't delete QgsRubberBand3D entities from this class
as they are owned by RubberBandRootEntity, which deletes them on windows
close. Only point rubberbands leave artifacts in scene whene deleted by
default destructor.

gacarrillor committed:
[ux] Stack diagrams drag'n'drop: clear current item after the drop event because it sticks to last activated item's index (the one that was dragged, thus remaining as a leftover from the user perspective, who would be more interested in the dropped index being the current item)

Withalion committed:
Keep 3D editing toolbar position on closing

uclaros committed:
Move layer setSubsetString() logic out of the point cloud query builder

uclaros committed:
Allow setting a filter expression to select points that should be edited

Withalion committed:
Finish editing when 2. point selected with line tool

wonder-sk committed:
Initial implementation of globe camera controller

This is a new navigation mode for QgsCameraController which is meant to work
just like the terrain-based navigation mode, but for 3D scenes with globe.

Initially, the new navigation mode can handle:
- zooming in/out with mouse wheel
- orbit around globe when dragging with left mouse button pressed
- left/right/up/down keys to orbit around globe
- shift + left/right keys to change heading angle
- shift + up/down keys to change pitch angle
- page up/page down keys to increase/decrease elevation

For camera pose, we use QgsCameraPose as for "flat" scenes, but the QCamera
is positioned/rotated slightly differently - the main change is that in case
of globe, we use ECEF coordinates of the camera pose's center point, convert
them to lat/lon coordinates in order to set up initial rotation of the camera
so that it is perpendicular to the tangent plane at the given ECEF coordinate.

JanCaha committed:
all temporary output on selection panel

JanCaha committed:
handle temporary output

ptitjano committed:
qgsprofilerenderer: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the curve's vertices.

This is achieved with mainly 2 changes:
- `QgsProfilePlotRenderer::setSubsectionsSymbol()` allows to set the
line symbol of the vertical lines
- `renderSubsectionsIndicator` generates and renders the vertical
lines. It is called by `QgsProfilePlotRenderer::renderToImage()`.

ptitjano committed:
qgsprofilerenderer: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the curve's vertices.

This is achieved with mainly 2 changes:
- `QgsProfilePlotRenderer::setSubsectionsSymbol()` allows to set the
line symbol of the vertical lines
- `renderSubsectionsIndicator` generates and renders the vertical
lines. It is called by `QgsProfilePlotRenderer::renderToImage()`.

ptitjano committed:
qgslayoutitemelevationprofile: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the curve's vertices.

This is achieved with mainly 2 changes:
- `setSubsectionsSymbol()` allows to set the line symbol of the
vertical lines
- Call `renderSubsectionsIndicator` on the `QgsProfilePlotRenderer`

ptitjano committed:
qgslayoutitemelevationprofile: Add support for subsections indicator

This adds support to display vertices lines over the profile graph to
display indicator at the location of the curve's vertices.

This is achieved with mainly 2 changes:
- `setSubsectionsSymbol()` allows to set the line symbol of the
vertical lines
- Call `renderSubsectionsIndicator` on the `QgsProfilePlotRenderer`

ptitjano committed:
qgsprofilerenderer: Remove empty line at the end of file

ptitjano committed:
qgsprofilerenderer: Remove empty line at the end of file

ptitjano committed:
qgslayoutelevationprofilewidget: Add support for subsections indicator

This adds some new customization associated with the introduction of
subsections indicator:
- it allows to display or hide them
- it allows to change the associated symbology

ptitjano committed:
qgslayoutelevationprofilewidget: Add support for subsections indicator

This adds some new customization associated with the introduction of
subsections indicator:
- it allows to display or hide them
- it allows to change the associated symbology

ptitjano committed:
qgslayoutitem: Remove empty lines at the end of file

ptitjano committed:
qgslayoutitem: Remove empty lines at the end of file

ptitjano committed:
qgslayoutitem: Remove empty lines at the end of file

lbartoletti committed:
Revert "fix(QgsMapToolCapture): Allow snapping point with different crs"

This reverts commit 7a0f84f39f2c412f6226b165d5f0b299284d1c4c.

lbartoletti committed:
fix: retrieve Z and M values in QgsMapToolCapture::nextPoint

Corrected logical error where Z and M values were not retrieved from layerPoint.
Additionally, ensured Z and M values are captured via match without reprojection
to prevent multiple transformations causing recessions.

Withalion committed:
Add maximum size for paintbrush tool

nyalldawson committed:
Polygonize: Don't assume geos results are always a multipolygon

nyalldawson committed:
polygonize: fix progress reporting

nyalldawson committed:
Fix some coverity move instead of copy warnings

nyalldawson committed:
Fix some (harmless) coverity nullptr dereference warnings

nyalldawson committed:
Fix possible divide by zero in raster rank algorithm

nyalldawson committed:
Fix incorrect error variable comparison

nyalldawson committed:
Store preferred http method for connections

nyalldawson committed:
Add synchronous argument to POST request methods

nyalldawson committed:
[wfs] Use POST requests when connection is set to prefer POST

Refs #32348

nyalldawson committed:
Fix build warnings

nyalldawson committed:
Silence false positive cpp check

ptitjano committed:
qgsprofilerenderer: Introduce a default symbol for subsections

This will be used in the next commit.

ptitjano committed:
qgsprofilerenderer: Introduce a default symbol for subsections

This will be used in the next commit.

nicogodet committed:
update options when widget is shown

ptitjano committed:
qgsprofilerenderer: Fix typo in header

ptitjano committed:
qgsprofilerenderer: Fix typo in header

alexbruy committed:
add "Add result to project" checkbox to mesh calculator dialog (fix #45245)

rouault committed:
[FEATURE] [OGC API provider] Implement Part 5 - Schemas

rouault committed:
[OGC API provider] Implement geometryColumnName() when Part 5 is available

nyalldawson committed:
Remove dead code, fix tests

nyalldawson committed:
Avoid loss of field settings when importing tables via browser

Disable the generic field type conversion (as is done elsewhere
in browser), as we have already explicitly set field types
and settings to match the destination provider

nyalldawson committed:
Minor cleanup

nyalldawson committed:
Default new field expressions to NULL

When adding a new field during a database table import, set the
initial expression for the field to "NULL" instead of an empty
string.

The empty string results in an error after accepting the dialog
("incomplete expression"), whereas NULL just creates an
unpopulated field, which better matches user expectations

nyalldawson committed:
Ensure Import Layer dialog is initially populated

Instead of requiring a change of source layer to populate

nyalldawson committed:
Use schema combo box in import vector layer dialog

Allows destination schema to be changed in the dialog, showing
choice of all available schemas

nyalldawson committed:
[mssql] Fix connection corruption after executing sql

Executing any SQL on a SQL Server connection would result in the
connection being dropped while it was still in use, leaving
the provider in a interdeterminate state which would often
refuse to run futher queries.

The QSqlQuery class does NOT have a virtual destructor, so we
need to explicitly use the QgsMssqlQuery class for the results
iterator so that the QgsMssqlQuery destructor gets called and
the shared reference to the QgsMssqlDatabase instance gets
released correctly.

nyalldawson committed:
Condense checks

nyalldawson committed:
[api] Add advanced network request preprocessor capabilities

Allows mutating the request, network operation and request data.

Intended for mocking eg post requests in unit tests

nyalldawson committed:
Add unit tests for WFS with post

nyalldawson committed:
Remove duplicate filter element

lbartoletti committed:
rewrite test

alexbruy committed:
add signal to notify about extent layer changes to QgsExtentWidget and
QgsExtentGroupBox

eltonhcchan committed:
Add protocol values for ESRI Geoportal (#61047)

* Add protocol values for ESRI Geoportal

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

alexbruy committed:
replace spinboxes with extent selector in the raster calculator dialog

alexbruy committed:
replace spinboxes with extent selector in the mesh calculator dialog

lbartoletti committed:
fix(system memory): Use robust GDAL version

lbartoletti committed:
prefer CPLGetUsablePhysicalRAM

Co-authored-by: Even Rouault <even.rouault@spatialys.com>

ptitjano committed:
qgspoint: Use QgsGeometryUtilsBase to compute azimuth

This way, the formula is only defined once.

nyalldawson committed:
[cleanup] Avoid redundant container lookups

nyalldawson committed:
[mssql] Use method which takes an existing connection instead of creating new connection

We already have a connection here:

nyalldawson committed:
[mssql] Optimise geom column type resolution

Don't create a new connection per table, when we can
just use one connection for all

nyalldawson committed:
More thread safety for QgsVectorLayerExporterTask

nyalldawson committed:
Fix leak after creating table on sql server

nyalldawson committed:
Fix crash when db import dialog closes

nyalldawson committed:
Move DOM compare function to QgsTestUtils

nyalldawson committed:
Just require ints instead of QNetworkAccessManager::Operation enum

The sip conversion code fails with enums under qt 6

nyalldawson committed:
[mssql] Remove semi-broken duplicate schema retrieval method

Avoids missing schemas in import vector layer combo box

nyalldawson committed:
Fix test for qt 6

Commit:
auto sipify 🍺

nyalldawson committed:
Add special handling for geocentric CRS in QgsCoordinateReferenceSystem::toGeographicCrs()

proj_crs_get_geodetic_crs considers a geocentric crs as already a
geographic CRS, whereas this method advertises that it will always
return a geographic latitude/longitude based CRS

So we build a geographic CRS using the same datum as the original CRS

lbartoletti committed:
fix warning: narrowing conversion from 'GIntBig' (aka 'long long') to signed type 'int'

JanCaha committed:
update strings

Co-authored-by: Nyall Dawson <nyall.dawson@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

JanCaha committed:
add types

alexbruy committed:
add line of sight svg arrow

lbartoletti committed:
fix(clang-tidy): narrowing conversion

mach0 committed:
adding austrian csw to metasearch default servers

ptitjano committed:
qgselevationprofiletoolmeasure: Factor out the logic to clear the tool

ptitjano committed:
qgselevationprofilewidget: Clear the measure tool on curve change

When a new profile curve is set, if measure tool need to be cleared if
it was active.

ptitjano committed:
qgselevationprofilewidget: Clear the measure tool on clear

rouault committed:
[Raster] Use HEIGHT_AVERAGE from RPC when available

rouault committed:
[GDAL] Deal with [C]Float16 of GDAL 3.11 as [C]Float32

Related to https://gdal.org/en/latest/development/rfc/rfc100_float16_support.html

nyalldawson committed:
Add transform logic tests for #60807

No regressions here, just differences in a nonsensical transform
of the entire globe in EPSG:3857 to a local projection EPSG:5514

nyalldawson committed:
Rename variables so its clear what CRS they are in

nyalldawson committed:
Fix missing render of global raster in small area local projection

If we are trying to view a global layer in a small area local
projection, then the transform of the map extent to the layer's
local projection can often fail.

Add fallback to try the opposite logic, and see if we can first
calculate the visible region in the layers extent, and then
transform that small region back to the map's crs

Fixes #60807

nyalldawson committed:
[mssql] Fix creation of geometry_columns table

The query was testing whether [dbo].[geometry_columns] exists,
but then attempting to create just [geometry_columns] (without
"dbo").

This fails if the table will be created in any schema other then
"dbo".

Adjust to check whether just [geometry_columns] exists, so that
we are checking whether the table exists in the ACTUAL schema
we'll be creating it in.

Fixes #57614

nyalldawson committed:
Add test masks

nyalldawson committed:
Remove unused dead code

Probably a remnant of copy/paste from the original implementation
from the postgres provider

nyalldawson committed:
[mssql] Fix native integer types

bigint should map to LongLong variant type

Add actual type names to display strings

nyalldawson committed:
[mssql] Fix fields get unaligned when importing data

troopa81 committed:
refactor(ServerProperties): write XML server properties in map layer

Withalion committed:
Fix deletion of 3D rubberbands

mjoerg committed:
[opencl] set source path when using qgis_process

This is necessary to execute OpenCL-enabled algorithms using qgis_process, when OpenClEnabled is true.

Commit:
auto sipify 🍺

DiGro committed:
Updated brand name SpatiaLite (capital L)

Line 257 : Spatialite Import = SpatiaLite Import
Line 280 : Spatialite Import = SpatiaLite Import

benoitdm-oslandia committed:
fix(fcgiserver): replace end-of-monitoring QgsDebugMsgLevel message by QgsMessageLog

benoitdm-oslandia committed:
fix(fcgiserver): replace end-of-monitoring QgsDebugMsgLevel message by QgsMessageLog

benoitdm-oslandia committed:
fix(fcgiserver): fix concurrent race condition to detect end of response

Remove duplicate "mShouldStop.store( false )" in "run()" as already set in constructor. In
the case when "stop()" is called before the "run()" (very very short request and system is
too slow to start the thread), the monitoring thread did not detect the end of response
and the request went in timeout.

benoitdm-oslandia committed:
fix(fcgiserver): fix concurrent race condition to detect end of response

Remove duplicate "mShouldStop.store( false )" in "run()" as already set in constructor. In
the case when "stop()" is called before the "run()" (very very short request and system is
too slow to start the thread), the monitoring thread did not detect the end of response
and the request went in timeout.

uclaros committed:
fix changing the paintbrush size using the wheel on precision mouse

uclaros committed:
fix changing the paintbrush size using the wheel on precision mouse

nicogodet commented:
Please read contribution guidelines.
You should target master branch.

Commit:
There were renamed labels' titles Connection to File name in qgsgeopackageprojectstoragedialog.ui,
Database to File name, Table name to Layer name in qgsnewgeopackagelayerdialogbase.ui.
Also there were renamed labels' names and edit fields' which are relating with these labels.
There was removed debuging string which was created for the test.

Commit:
There were renamed labels' titles Connection to File name in qgsgeopackageprojectstoragedialog.ui,
Database to File name, Table name to Layer name in qgsnewgeopackagelayerdialogbase.ui.
Also there were renamed labels' names and edit fields' which are relating with these labels.

Snegovoy98 commented:
Ok. Thanks.

Commit:
There was removed debuging string which was created for the test.

Snegovoy98 commented:
Yes of course.

Snegovoy98 commented:
@nicogodet, I was reworded commits.

nyalldawson commented on src/ui/qgsgeopackageprojectstoragedialog.ui:
```diff
@@ -16,12 +16,12 @@
       <widget class="QLabel" name="label">
        <property name="text">
-        <string>Connection</string>
+        <string>File name</string>
```
This combo shows not just filenames, but also the names of stored geopackage connections. I don't think using "file name" is appropriate here.

nyalldawson committed:
Address review

wonder-sk committed:
Add 3D scene mode enum

wonder-sk committed:
Review from Stefanos and Nyall

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

wonder-sk committed:
fix tests for 4978 rendering

updated the expected images because the terrain based
camera controller no longer applies to globe views

alexbruy commented on src/ui/qgsnewgeopackagelayerdialogbase.ui:
```diff
@@ -437,19 +437,19 @@
+          <widget class="QLabel" name="mLayerNameLabel">
            <property name="text">
-            <string>Table name</string>
+            <string>Layer name</string>
```
It is possible to create plain tables too. Also all layers are in fact also tables with special columns and associated entries in the system tables.

gacarrillor committed:
[fix] Make stacked diagrams behave as expected with horizontal fixed-size stacked bars (#61082)

wonder-sk committed:
add test image for qt6

wonder-sk committed:
make clang-tidy happy

uclaros committed:
Set minimum width back to 5 and make clang-tidy happy

uclaros committed:
Set minimum width back to 5 and make clang-tidy happy

benoitdm-oslandia committed:
fix(ci/ogc): move log outputs from file to stderr as logs are not displayed and file is not used.

benoitdm-oslandia committed:
fix(ci/ogc): move log outputs from file to stderr as logs are not displayed and file is not used.

jmonticolo committed:
Add order by enumsortorder in the query for enum values

wonder-sk committed:
fix second 4978 control image

jef-n committed:
release.pl: don't error out when a translation is dropped (because it falls below the 35% threshold)

(cherry picked from commit d363994a6480607d2d3ead3e41527cded38688de)

Snegovoy98 commented:
@nyalldawson, @alexbruy thanks for your review. Tell me please am I need to rename these fields how they were or  I need to think about suitable names and rename them?

ptitjano committed:
testqgs3drendering: Disable opacity for the polygons clipping test

This will make it easier to test enabling/disabling clipping multiple
times with the other reference image.

ptitjano committed:
testqgs3drendering: Improve coverage for plane clipping

This tests that enabling and disabling clipping multiple times works.

ptitjano committed:
testqgspointcloud3drendering: Improve coverage for plane clipping

This tests that enabling and disabling clipping multiple times works.

Withalion committed:
qgsmaterial: Fix removal of parameters when clipping is disabled

Plane clipping uses two `QParameters` named
`CLIP_PLANE_ARRAY_PARAMETER_NAME` and
`CLIP_PLANE_MAX_PLANE_PARAMETER_NAME`. When clipping is disabled both
parameters need to be removed to prevent side effects on the shader
code. However, the loops which iterates through the parameters breaks
as soon one parameter is found. This means that the second parameter
is never removed.

This issue is fixed by removing the `break` condition. This way, both
parameters are always removed from the material.

ptitjano committed:
testqgs3dmaterial: Add tests for clipping

Commit:
Merge branch 'master' into fix_issue_60990

ptitjano committed:
qgsline3dsymbol_p: Use qgsgeometry_cast

ptitjano committed:
qgsline3dsymbol_p: Use qgsgeometry_cast

ptitjano committed:
qgsline3dsymbol_p: Give proper name to some variables

ptitjano committed:
qgsline3dsymbol_p: Give proper name to some variables

ptitjano committed:
qgsline3dsymbol_p: Fix some clang-tidy warnings

ptitjano committed:
qgsline3dsymbol_p: Fix some clang-tidy warnings

ptitjano committed:
qgsline3dsymbol_p: Fix some clang-tidy warnings

wonder-sk committed:
Fix tiled scene chunk loader for globe scenes

This disables the workarounds for nodes of 3D tiles with large spatial extent
when the scene is shown as globe, because in case of globe those workarounds
are not needed, and actually break the functionality.

With this fix, google 3D tiles work with globe :tada:

jef-n committed:
no gettid() on OSX (followup 154ad712e9c25d55a5afcaa221e99016b9db29df)

jef-n committed:
update_ts.sh: alternatively use pylupdate6 + python translation string fixes

Commit:
There were rollbacked field File Name to Connection in qgsgeopackageprojectstoragedialog.cpp, Layer Name to Table name in qgsnewgeopackagelayerdialog

Commit:
There were rollbacked field File Name to Connection in qgsgeopackageprojectstoragedialog.cpp, Layer Name to Table name in qgsnewgeopackagelayerdialog

nyalldawson committed:
Expand docs

nyalldawson committed:
Explain raw value

Commit:
auto sipify 🍺

nyalldawson committed:
Don't write empty metadataUrls, keywords to XML

nyalldawson committed:
Don't write empty metadataUrls, keywords to XML

YoannQDQ committed:
Fix #60968 - Duplicated layer variables

YoannQDQ committed:
Fix #60968 - Duplicated layer variables

YoannQDQ committed:
Add test

YoannQDQ committed:
Add test

Commit:
auto sipify 🍺

jef-n committed:
verify paths are within the zip directory

jef-n committed:
verify paths are within the zip directory

jef-n committed:
verify paths are within the zip directory

ptitjano committed:
qgstessellator: Do not use a custom function to compute ring orientation

ptitjano committed:
qgstessellator: Do not use a custom function to compute ring orientation

Commit:
Merge branch 'master' into fix_issue_60990

Commit:
auto sipify 🍺

Commit:
auto sipify 🍺

nyalldawson committed:
Slightly more helpful error message when OGR attribute creation fails

Include the attempted field type in the error message, so that it's
clear what type we were attempting to create. This is useful when
we get an error like:

"Creation of field fid (QString) failed (OGR error: Wrong field
type for fid)"

nyalldawson committed:
Slightly more helpful error message when OGR attribute creation fails

Include the attempted field type in the error message, so that it's
clear what type we were attempting to create. This is useful when
we get an error like:

"Creation of field fid (QString) failed (OGR error: Wrong field
type for fid)"

nyalldawson committed:
Fetch subtype once only

nyalldawson committed:
Fetch subtype once only

nyalldawson committed:
[ogr] Fix crash when adding feature

If fields are not set for the feature, and a field value is invalid
for the actual layer's type we would get a crash on debug enabled builds

nyalldawson committed:
[ogr] Fix crash when adding feature

If fields are not set for the feature, and a field value is invalid
for the actual layer's type we would get a crash on debug enabled builds

nyalldawson committed:
Add splitter to Execute sql window

Allow resizing the query editor/results table, and remember
sizes

nyalldawson committed:
Add splitter to Execute sql window

Allow resizing the query editor/results table, and remember
sizes

alexbruy reviewed (changes requested):
Can you please fix tests and squash?

rouault committed:
[Raster] Tune PR #61091 to rename HEIGHT_AVERAGE as HEIGHT_DEFAULT, to follow latest change of https://github.com/OSGeo/gdal/pull/11989

rouault committed:
[Raster] Tune PR #61091 to rename HEIGHT_AVERAGE as HEIGHT_DEFAULT, to follow latest change of https://github.com/OSGeo/gdal/pull/11989

Snegovoy98 commented:
> Can you please fix tests and squash?

Tell me please, what am I need to squash?

Commit:
There were renamed Database to File Name in qgsnewgeopackagelayerdialogbase.ui, renamed mDatabase to mFileName in
src/gui/qgsnewgeopackagelayerdialog.cpp, src/gui/qgsnewgeopackagelayerdialog.h and  merged master

Snegovoy98 commented:
> > Tell me please, what am I need to squash?
> 
> Your commits.

I squashed commits.

Commit:
Merge branch 'master' into fix_issue_60990

Commit:
Merge branch 'fix_issue_60990' of https://github.com/Snegovoy98/QGIS into fix_issue_60990

Commit:
Merge branch 'master' into fix_issue_60990

Commit:
Merge branch 'fix_issue_60990' of https://github.com/Snegovoy98/QGIS into fix_issue_60990

Commit:
Merge branch 'master' into fix_issue_60990

Commit:
Merge branch 'fix_issue_60990' of https://github.com/Snegovoy98/QGIS into fix_issue_60990

Commit:
Merge branch 'master' into fix_issue_60990

Snegovoy98 commented:
[alexbruy](https://github.com/alexbruy), can you review again?

Commit:
Merge branch 'master' into fix_issue_60990

Commit:
There was fix merge conflict

Commit:
There was merged master

Commit:
There was merged master

Commit:
There was merged master

Commit:
There was merged master

Commit:
There was merged master

Snegovoy98 committed:
Update qgsprofilerenderer.sip.in

Snegovoy98 committed:
Update qgsprofilerenderer.sip.in

There was fixed conflict.

nyalldawson commented:
@Snegovoy98 there's bad merges here, you've pulled in unrelated changes.

In general I'd advise NOT to continously merge update like this. It's not necessary, and it's just more work for you + it clutters up the git history

Snegovoy98 commented:
> @Snegovoy98 there's bad merges here, you've pulled in unrelated changes.
> 
> In general I'd advise NOT to continously merge update like this. It's not necessary, and it's just more work for you + it clutters up the git history
@nyalldawson tell me please, how can I fixed this merge?
